src/convertToObject.js

```diff
@@ -3,16 +3,33 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const minPropertyLength = 5;
+  const splittedStyle = sourceString.split('\n');
+  const formatedStyle = {};
+
+  splittedStyle
+    .filter(property => property.length > minPropertyLength)
+    .map(property => {
+      const splitted = property.trim().split(':');
+
+      return splitted.map(key => {
+        return key.replace(';', '').trim();
+      });
+    })
+    .forEach(property => {
+      formatedStyle[property[0]] = property[1];
+    });
```

Comment:
You don't need 3 iterations to do this. Let's improve this algorithm and make the same just in one iteration